include/range/v3/algorithm/shuffle.hpp

```diff
@@ -76,6 +76,16 @@ namespace ranges
                 return orig;
             }
 
+            template<typename I, typename S,
+                CONCEPT_REQUIRES_(RandomAccessIterator<I>() && IteratorRange<I, S>() &&
+                    Permutable<I>())>
+            I operator()(I begin, S end_) const
+            {
+                std::random_device rng;
+                std::mt19937 urng(rng());
```

Comment:
Keep it simple, please. We can complexify it later if there is a demonstrated need.